doc/userguide/devguide/internals/pipeline/packet-pipeline.rst

```diff
@@ -0,0 +1,51 @@
+===============
+Packet Pipeline
+===============
+
+Main Takeaways:
+
+- What is Suricata's Packet Pipeline
+- Main components
+- How do these work together
+- Different thread models (?)
+
+Introduction
+============
+
+The Packet Pipeline might be considered the core part of Suricata. It is in this
+macro-component that the Packet processing happen (``PacketProcessing``
+threads). The two other main components are **Management threads**
+responsible for *Flow Managing* and *Stats*.
+
+In Suricata, the packet pipeline starts with the Packet Capture.
+
+The pipeline either runs in a single thread (runmode *single* or *workers*) or
+split in 2 stages (*autofp*).
+
+A pipeline is created by Suricata's ``RunMode`` and stored in per thread
```

Comment:
runmodes are defined by the capture method (e.g. runmode-af-packet.c), but generally implement only 3 predefined modes: workers, autofp, single